### PR TITLE
RC-22 Download MP3

### PR DESCRIPTION
**Pull request by miltonleung**

### CODE REVIEW

File: RocketCast/DownloadBridge.swift [hunk 1]
```diff
@@ -9,12 +9,13 @@
...
 import Foundation
 import UIKit
+import AVFoundation
 
 protocol DownloadBridgeProtocol {
    
     func downloadPodcastXML(url:PodcastWebURL, result:(url: PodcastStorageURL?) -> ())
     func downloadImage(url: ImageWebURL, result:(url: ImageStorageURL) -> ())
-    func downloadMp3(url: MP3WebURL, result:(url: MP3StorageURL) -> ())
+    func downloadAudio(url: AudioWebURL, result:(url: AudioStorageURL?) -> ())
```
Comment (mohamedali92) [inline RocketCast/DownloadBridge.swift]: Why did you rename Mp3 to Audio?

File: RocketCastTests/DownloadTests.swift [hunk 2]
```diff
@@ -99,11 +102,90 @@ class DownloadTests: XCTestCase {
         waitUntil(6) {done}
     }
     
+    func testDownloadPodcastAudio() {
+        // Test real audio
+        var done = false
+        
+        var url = "http://www.scientificamerican.com/podcast/podcast.mp3?fileId=14824345-7D79-454F-9A8F30B98EE219F3"
```
Comment (mohamedali92) [inline RocketCastTests/DownloadTests.swift]: what happens if this website goes down?

File: RocketCastTests/DownloadTests.swift [hunk 3]
```diff
@@ -99,11 +102,90 @@ class DownloadTests: XCTestCase {
...
+            
+            if let data = fileMgr.contentsAtPath(path) {
+                XCTAssertEqual(data.length, 2002733)
+            }
+        })
+        
+        waitUntil(6) {done}
+        done = false
+        
+        url = "http://www.scientificamerican.com/podcast/podcast.mp3?fileId=C13A2C3C-F951-4C81-BF7B323A1D0C5A28"
```
Comment (mohamedali92) [inline RocketCastTests/DownloadTests.swift]: Why are you testing the same thing twice?

File: RocketCast/DownloadBridge.swift (hunk 1 above)
Comment (miltonleung) [inline RocketCast/DownloadBridge.swift]: because not all podcasts are in mp3 format, so i chose a more general term

File: RocketCast/DownloadBridge.swift [hunk 4]
```diff
@@ -61,8 +62,37 @@ extension ModelBridge: DownloadBridgeProtocol {
         //TODO
     }
     
-    func downloadMp3(url: MP3WebURL, result:(url: MP3StorageURL) -> ()) {
-        //TODO
+    func downloadAudio(url: AudioWebURL, result:(url: AudioStorageURL?) -> ()) {
+        let audioURL = NSURL(string: url)
+
```
Comment (miltonleung) [inline RocketCast/DownloadBridge.swift]: i removed white space, is there something more specific you meant to say?

File: RocketCast/PlayerView.xib [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="10117" systemVersion="15F34" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="10116" systemVersion="15G1004" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES">
```
Comment (miltonleung) [inline RocketCast/PlayerView.xib]: oops, it didn't really change anything though

File: RocketCastTests/DownloadTests.swift [hunk 6]
```diff
@@ -45,13 +47,15 @@ class DownloadTests: XCTestCase {
             XCTAssertNotNil(downloadedPodcast)
             
             let path = NSHomeDirectory().stringByAppendingString(downloadedPodcast!)
-            
+            print(path)
```
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: deleted

File: RocketCastTests/DownloadTests.swift [hunk 7]
```diff
@@ -60,6 +64,7 @@ class DownloadTests: XCTestCase {
             XCTAssertNotNil(downloadedPodcast)
             
             let path = NSHomeDirectory().stringByAppendingString(downloadedPodcast!)
+            print(path)
```
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: deleted

File: RocketCastTests/DownloadTests.swift (hunk 3 above)
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: its a different podcast

File: RocketCastTests/DownloadTests.swift (hunk 2 above)
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: i think thats sort of an extraneous case, so that why i made two tests

File: RocketCastTests/DownloadTests.swift [hunk 8]
```diff
@@ -45,13 +47,15 @@ class DownloadTests: XCTestCase {
...
             if let data = fileMgr.contentsAtPath(path) {
                 let xmlString = String(data:data, encoding: NSUTF8StringEncoding)!
                 Log.info(xmlString)
                 XCTAssertTrue(xmlString.containsString("xml") && xmlString.containsString("xml"))
             }
         }
+        waitUntil(6) {done}
+        done = false
         let url2 = "http://www.npr.org/rss/podcast.php?id=510298"
```
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: i think thats sort of an extraneous case, so that why i made two tests

File: RocketCastTests/DownloadTests.swift [hunk 9]
```diff
@@ -122,13 +188,25 @@ class DownloadTests: XCTestCase {
...
+//        var error:NSError?
+//        do {
+//            try audioPlayer = AVAudioPlayer(data: data)
+//        } catch let error as NSError {
+//            Log.error(error.debugDescription)
+//        }
+//        audioPlayer.prepareToPlay()
+//        audioPlayer.play()
+//    }
 }
```
Comment (QuantumSpark) [inline RocketCastTests/DownloadTests.swift]: Remove this commented out code
